CHANGELOG.md

```diff
@@ -4,9 +4,77 @@ This document includes a curated changelog for each release. We also publish a c
 a [GitHub release](https://github.com/nginxinc/nginx-gateway-fabric/releases), which, by contrast, is auto-generated
 and includes links to all PRs that went into the release.
 
+## Release 1.3.0
+
+_June 11, 2024_
+
+FEATURES:
+
+- The minimum supported version of Kubernetes is now 1.25. [1885](https://github.com/nginxinc/nginx-gateway-fabric/pull/1885)
```

Comment:
The minimum supported version of Kubernetes is now 1.25. 
and
 NGINX Plus was updated to R32.

don't really look like features.

Perhaps we can have CHANGES section for things like that?

Reply:
CHANGES makes me think of breaking changes, but I agree that they're not really features and should probably go somewhere else

Reply:
Moved these and the gw api bump to a new DEPENDENCIES section